src/scene/vcPOI.cpp

```diff
@@ -771,3 +963,122 @@ bool vcPOI::LoadAttachedModel(const char *pNewPath)
 
   return false;
 }
+
+bool vcPOI::IsVisible(vcState *pProgramState)
+{
+  // if POI is invisible or if it exceeds maximum visible POI distance (unless selected)
+  bool visible = m_visible;
+  visible = visible && m_selected;
+  visible = visible && (udMag3(m_pLabelInfo->worldPosition - pProgramState->camera.position) < pProgramState->settings.presentation.POIFadeDistance);
```

Comment:
Should these be `||`?

Reply:
You're right, this is incorrect. A POI is visible if it is _not_ invisible _and_ it must be in the view frustum, _but_ if it is selected it should be visible.